[BUG] Compile error using gcc 13.2.0

**Describe the bug**
The branch-24.10 has the following compile error when built with gcc 13.2.0:
```
In file included from /home/biubiuty/biubiu/rmm/rmm/tests/mr/device/adaptor_tests.cpp:26:
/home/biubiuty/biubiu/rmm/rmm/include/rmm/mr/device/logging_resource_adaptor.hpp:292:38: error: attribute ignored [-Werror=attributes]
  292 |   friend logging_resource_adaptor<T> make_logging_adaptor(T* upstream,
      |                                      ^~~~~~~~~~~~~~~~~~~~
/home/biubiuty/biubiu/rmm/rmm/include/rmm/mr/device/logging_resource_adaptor.hpp:292:38: note: an attribute that appertains to a friend declaration that is not a definition is ignored
cc1plus: all warnings being treated as errors
make[2]: *** [tests/CMakeFiles/ADAPTOR_TEST.dir/build.make:76: tests/CMakeFiles/ADAPTOR_TEST.dir/mr/device/adaptor_tests.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1327: tests/CMakeFiles/ADAPTOR_TEST.dir/all] Error 2
make: *** [Makefile:166: all] Error 2
b
```

**Steps/Code to reproduce bug**
Use gcc version 12.1 or newer, and follow the instruction to build RMM from source.

**Expected behavior**
No compile error.

**Environment details (please complete the following information):**
 - Environment location: Bare-metal
 - Method of RMM install: from source
 - Please run and attach the output of the `rmm/print_env.sh` script to gather relevant environment details

[print_env.txt](https://github.com/user-attachments/files/16896733/print_env.txt)



**Additional context**
N/A

--- a/include/rmm/mr/device/logging_resource_adaptor.hpp
+++ b/include/rmm/mr/device/logging_resource_adaptor.hpp
@@ -283,16 +283,6 @@ class logging_resource_adaptor final : public device_memory_resource {
     return get_upstream_resource() == cast->get_upstream_resource();
   }
 
-  // make_logging_adaptor needs access to private get_default_filename
-  template <typename T>
-  // NOLINTNEXTLINE(readability-redundant-declaration)
-  [[deprecated(
-    "make_logging_adaptor is deprecated in RMM 24.10. Use the logging_resource_adaptor constructor "
-    "instead.")]]
-  friend logging_resource_adaptor<T> make_logging_adaptor(T* upstream,
-                                                          std::string const& filename,
-                                                          bool auto_flush);
-
   std::shared_ptr<spdlog::logger> logger_;  ///< spdlog logger object
 
   Upstream* upstream_;  ///< The upstream resource used for satisfying

